components/plan-build/bin/public.bash

```diff
@@ -452,8 +457,9 @@ fix_interpreter() {
     local pkg=$2
     local int=$3
     local interpreter_old=".*${int}"
-    local interpreter_new="$(pkg_interpreter_for ${pkg} ${int})"
-    if [[ -z $interpreter_new || $? -ne 0 ]]; then
+    local interpreter_new
+    # shellcheck disable=2154
+    if ! interpreter_new="$(pkg_interpreter_for "${pkg}" "${int}")" || [[ -z $interpreter_new ]]; then
```

Comment:
```
In ./components/plan-build/bin/public.bash line 456:
    if [[ -z $interpreter_new || $? -ne 0 ]]; then
                                 ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```
I believe this is the appropriate fix. See https://github.com/koalaman/shellcheck/wiki/SC2181